File: backend/src/main/java/com/smarttravel/services/BookingService.java
package com.smarttravel.services;

import com.smarttravel.dto.request.BookingCreateRequest;
import com.smarttravel.dto.response.BookingResponse;

import java.math.BigDecimal;
import java.util.List;

public interface BookingService {
    BookingResponse createBooking(BookingCreateRequest request, String userEmail);
    List<BookingResponse> getMyBookings(String userEmail);
    List<BookingResponse> getAllBookings();
    BookingResponse getBookingById(Long id, String userEmail);
    BookingResponse cancelBooking(Long id, String userEmail);
    BookingResponse updateBookingStatus(Long id, com.smarttravel.enums.BookingStatus status);
    BigDecimal validateAndCalculateVoucher(String voucherCode, BigDecimal originalTotal);
}

File: frontend/src/services/bookingService.ts
import axiosClient from './axiosClient';
import { Booking, BookingCreateRequest } from '../types/booking';
import { ApiResponse } from '../types/common';

export const bookingService = {
  createBooking: async (data: BookingCreateRequest): Promise<ApiResponse<Booking>> => {
    return await axiosClient.post('/bookings', data);
  },

  getMyBookings: async (): Promise<ApiResponse<Booking[]>> => {
    return await axiosClient.get('/bookings/my-bookings');
  },

  getBookingById: async (id: number): Promise<ApiResponse<Booking>> => {
    return await axiosClient.get(`/bookings/${id}`);
  },

  cancelBooking: async (id: number): Promise<ApiResponse<Booking>> => {
    return await axiosClient.post(`/bookings/${id}/cancel`);
  },

  updateBookingStatus: async (id: number, status: string): Promise<ApiResponse<Booking>> => {
    return await axiosClient.put(`/bookings/${id}/status?status=${status}`);
  },

  getVendorBookings: async (): Promise<ApiResponse<Booking[]>> => {
    return await axiosClient.get('/vendor/bookings');
  },

  validateVoucher: async (code: string, originalTotal: number): Promise<ApiResponse<{ voucherCode: string; discountAmount: number; finalTotal: number }>> => {
    return await axiosClient.post('/bookings/validate-voucher', { code, originalTotal });
  },
};
